lua/treesitter-context/render.lua

```diff
@@ -20,15 +22,43 @@ local retired_buffers = {}
 local window_contexts = {}
 
 --- @return integer buf
-local function create_buf()
+local function create_or_get_buf()
+  for index=#buffers_pool, 1, -1 do
+    local buf = buffers_pool[index]
+    if api.nvim_buf_is_valid(buf) then
+      table.remove(buffers_pool, index)
```

Comment:
Shouldn't we remove the buffer if it's invalid?

Reply:
Yes, let's remove it. It should be for free as we remove the end of the array.